yt/geometry/particle_geometry_handler.py

```diff
@@ -46,6 +46,9 @@ def convert(self, unit):
         return self.dataset.conversion_factors[unit]
 
     def _setup_filenames(self):
+        if hasattr(self, "data_files"):
+            pass
```

Comment:
I think this should be "return" or something -- a `pass` here will still call the stuff below.

Reply:
Oops, you're totally right! For what it's worth, there do seem to be some instances (unrelated to this PR) where this gets called more than once. Maybe we can put a pin in that. We could also convert `data_files` to a property that calls `_setup_filenames`.